proposals/src/mappings/mappingHandlers.ts

```diff
@@ -134,50 +150,73 @@ export async function handlePropose(
   let proposalNumber: number;
   try {
     proposalNumber = Number(
-      findAttribute(JSON.parse(cosmosMessage.tx.tx.log), "wasm", "proposal_id")
-        .value
+      findAttribute(JSON.parse(log), "wasm", "proposal_id").value
     );
+    if (isNaN(proposalNumber)) {
+      throw new Error(
+        `proposalNumber (${JSON.stringify(proposalNumber)}) is NaN`
+      );
+    }
   } catch (err) {
     logger.error(
-      `Error retrieving proposalNumber for ${contract}: ${
+      `----- ${contract} > ${proposalNumber} ==> Failed to retrieve proposalNumber during propose: ${
         err instanceof Error ? err.message : `${err}`
       }`
     );
     return;
   }
 
   // Make proposal (creates module as well).
-  await updateOrCreateAndGetProposal(contract, proposalNumber, true, true);
-
-  logger.info(`----- ${contract} > ${proposalNumber} ==> Proposed`);
+  if (
+    await updateOrCreateAndGetProposal(
+      contract,
+      proposalNumber,
+      true,
+      new Date(header.time)
+    )
+  ) {
+    logger.info(`----- ${contract} > ${proposalNumber} ==> Proposed`);
+  } else {
+    logger.info(
+      `----- ${contract} > ${proposalNumber} ==> Failed during propose`
+    );
+  }
 }
 
-export async function handleVote(
-  cosmosMessage: CosmosMessage<DecodedMsg>
-): Promise<void> {
-  const {
-    contract,
-    sender,
-    msg: { vote },
-  } = cosmosMessage.msg.decodedMsg;
-
+export async function handleVote({
+  block: {
+    block: { header },
+  },
+  tx: {
+    tx: { log },
+  },
+  msg: {
+    decodedMsg: {
+      contract,
+      sender,
+      msg: { vote },
+    },
+  },
+}: CosmosMessage<DecodedMsg>): Promise<void> {
   // cw-proposal-single and cw-proposal-multiple supported
-  const voteKeys = Object.keys(vote).sort();
-  if (voteKeys.join(".") !== "proposal_id.vote") {
+  if (Object.keys(vote).sort().join(".") !== "proposal_id.vote") {
     return;
   }
 
-  const blockDate = new Date(cosmosMessage.block.block.header.time);
+  const blockDate = new Date(header.time);
   const proposalNumber = Number(vote.proposal_id);
 
   let proposalOpen: boolean;
   try {
-    proposalOpen =
-      findAttribute(JSON.parse(cosmosMessage.tx.tx.log), "wasm", "status")
-        .value === "open";
+    const status = findAttribute(JSON.parse(log), "wasm", "status").value;
```

Comment:
think you should be taking a message index here as well to make sure you're grabbing the right events if there are multiple messages in this transaction.